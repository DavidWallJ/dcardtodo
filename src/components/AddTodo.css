.todoInput {
	margin-top: 10px;
	margin-bottom: 24px;
}
