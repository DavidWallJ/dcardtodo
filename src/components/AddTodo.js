import React, { Component } from 'react';
import { Button, FormGroup, FormControl } from 'react-bootstrap';

class AddTodo extends Component {
	constructor() {
		super();

		this.state = { todoText: '' };
	}
	handleSubmit = e => {
		e.preventDefault();

		if (this.state.todoText.length > 0) {
			this.setState({ todoText: '' });
			this.props.onAddTodo(this.state.todoText);
		} else {
			// puts cursor back in text input field if no text exists on submit
			this.todoTextRef.focus();
		}
	};
	render() {
		return (
			<form onSubmit={this.handleSubmit.bind(this)}>
				<FormGroup>
					<FormControl
						className="todoInput"
						type="text"
						placeholder="...add your todo here"
						onChange={event => this.setState({ todoText: event.target.value })}
						// ref to move cursor back to if field is empty when submitted
						inputRef={input => (this.todoTextRef = input)}
					/>
					<Button bsStyle="primary" bsSize="large" type="submit" block>
						Add Todo
					</Button>
				</FormGroup>
			</form>
		);
	}
}

export default AddTodo;
