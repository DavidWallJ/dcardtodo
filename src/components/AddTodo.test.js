import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddTodo from './AddTodo';

describe(AddTodo, () => {
	const mockOnAddTodo = jest.fn();
	const component = shallow(<AddTodo onAddTodo={mockOnAddTodo} />);

	it('should exist', () => {
		const component = renderer.create(<AddTodo />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('contains the form', () => {
		expect(component.find('form')).toHaveLength(1);
		expect(component.find('Button')).toHaveLength(1);
		expect(component.find('.todoInput')).toHaveLength(1);
	});

	describe('when typing into the person `todoInput`', () => {
		const todoText = 'Buy milk';

		beforeEach(() => {
			component
				.find('.todoInput')
				.simulate('change', { target: { value: todoText } });
		});

		it('updates the `todoText` value', () => {
			expect(component.state().todoText).toEqual(todoText);
		});

		// it('calls the passed in `onAddTodo` function when add button is clicked', () => {
		// 	component.find('Button').simulate('click');
		// 	expect(mockOnAddTodo).toBeCalled();
		// });
	});
});
