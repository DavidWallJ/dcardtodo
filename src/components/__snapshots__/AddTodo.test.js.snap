// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class AddTodo extends _react.Component {
	constructor() {
		super();this.



		handleSubmit = e => {
			e.preventDefault();

			if (this.state.todoText.length > 0) {
				this.setState({ todoText: '' });
				this.props.onAddTodo(this.state.todoText);
			} else {
				// puts cursor back in text input field if no text exists on submit
				this.todoTextRef.focus();
			}
		};this.state = { todoText: '' };}
	render() {
		return (
			_react2.default.createElement('form', { onSubmit: this.handleSubmit, __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this },
				_react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this },
					_react2.default.createElement(_reactBootstrap.FormControl, {
						className: 'todoInput',
						type: 'text',
						placeholder: '...add your todo here',
						onChange: event => this.setState({ todoText: event.target.value })
						// ref to move cursor back to if field is empty when submitted
						, inputRef: input => this.todoTextRef = input, __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this }),

					_react2.default.createElement(_reactBootstrap.Button, { bsStyle: 'primary', bsSize: 'large', type: 'submit', block: true, __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }, 'Add Todo'))));





	}} should exist 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="todoInput form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="...add your todo here"
      type="text"
    />
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={false}
      type="submit"
    >
      Add Todo
    </button>
  </div>
</form>
`;
